validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -474,6 +488,23 @@ def stop(self):
                 # Connection has already been disconnected.
                 pass
 
+    def _request_chain_head(self):
+        peers = self._gossip.get_peers()
+
+        peered_connecions = [conn_id for conn_id in peers
+                             if self._connection_statuses[conn_id] ==
+                             PeerStatus.PEER]
+
+        if not peered_connecions:
+            return
+
+        LOGGER.debug('Have not received a chain head from peers. '
+                     'Requesting from %s',
+                     peered_connecions)
+
+        for conn_id in peered_connecions:
+            self._gossip.send_block_request("HEAD", conn_id)
```

Comment:
This request already happens whenever peering with a potential peer is successful: https://github.com/hyperledger/sawtooth-core/blob/master/validator/sawtooth_validator/gossip/gossip.py#L774

With that context, my interpretation of this change is that a node will now repeatedly send HEAD requests to the peers it has already successfully connected to until it has a chain head, while it is in dynamic peering. Is that correct?

Reply:
If so, can the other request for HEAD be removed now, since it will happen through this channel now?

Reply:
No - the added channel is only for the degenerate case where a node has no chain, and neither do its immediate peers.  

This initial request is still valid, as it still would need to query peers when reconnecting to a network (e.g. after it disconnected or crashed) - it has a chain, but it may be out-of-date.